apache/metron
# METRON-1782 Add Kafka Partition and Offset to Profiler Debug Logs

I was troubleshooting an issue that a user was running into with the Profiler.  The logs made it appear that a second batch of messages were being sent after a period of time, when none actually were. 

We found that the second batch was actually just Storm timing out the first batch of messages and re-sending them.  This would have been simple to uncover if we had the Kafka topic, partition, and offset details of each message logged at a DEBUG level.

I made those code change while debugging the issue.  I thought this might be generally useful to roll into Apache.

On a side note, I think it would be generally very useful to carry forward the topic/partition/offset/timestamp information for each message in Metron.  When an error occurs with a message, we can log this specific information which makes it easy to trace the source of the problem.  Right now, we typically log the entire message, which is really not grokable and not useful for tracing the source of the problem as many messages received will look either similar or exactly the same.

## Testing

1. Build Metron (this excludes building the UI code to save time.)
    ```
    mvn clean install -DskipTests -T2C -pl '!metron-interface/metron-config,!metron-interface/metron-alerts'
    ```

1. Alter `metron-analytics/metron-profilers/src/test/resources/log4j.properties` and uncomment the lines to turn on DEBUG logs for the Profiler.
    ```
    # uncomment below to help debug tests
    log4j.logger.org.apache.metron.profiler=ALL
    log4j.logger.org.apache.storm.windowing=ALL
    ```

1. Run the `ProfilerIntegrationTest` and review the log messages.  You should see a message like this.
    ```
    cd metron-analytics/metron-profiler
    mvn integration-test
    ```

1. You should see log messages like the following.  This shows us exactly which messages from the topic are being processed.

    ```
    DEBUG ProfileSplitterBolt:151 - Received message; topic=indexing, partition=0, offset=0, kafkaTimestamp=1537276089966
    ...
    DEBUG ProfileSplitterBolt:151 - Received message; topic=indexing, partition=0, offset=1, kafkaTimestamp=1537276090197
    ...
    DEBUG ProfileSplitterBolt:151 - Received message; topic=indexing, partition=0, offset=2, kafkaTimestamp=1537276090405
    ```

## Pull Request Checklist
